transformer_engine/jax/flax/transformer.py

```diff
@@ -1078,6 +1084,8 @@ def __call__(self,
         else:
             mha_name = 'self_attention'
 
+        inputs = _with_sharding_constraint(inputs, (BATCH_AXES, SEQLEN_AXES, HIDDEN_AXES))
```

Comment:
Do you want to include those WSC in this PR or this belong to another PR?

Reply:
Yes, adding WSC has been in TODO for a while, not regarding to CP.